Sample.java

```diff
@@ -1,7 +1,71 @@
-// Time Complexity :
-// Space Complexity :
-// Did this code successfully run on Leetcode :
-// Any problem you faced while coding this :
 
 
-// Your code here along with comments explaining your approach
+class Solution {
+
+    // Time Complexity : O(2^n)
+    // Space Complexity : O(n) 
+    // Did this code successfully run on Leetcode : No : Time Limit Exceeded
+    // Any problem you faced while coding this : No dp solution
+
+
+    // Paint house solution
+
+    public int minCost(int[][] costs) {
+        
+        if(costs == null || costs.length == 0)
+            return 0;
+        
+        return Math.min(helper(costs, 0, 0), Math.min(helper(costs, 0, 1), helper(costs, 0, 2)));
+    }
+    
+    private int helper(int[][] costs, int i, int j) {
+        
+        int price = costs[i][j];
+        if(i < costs.length-1) {
+            if(j == 0) {
+                price += Math.min(helper(costs, i+1, 1), helper(costs, i+1, 2));
+            } 
+            else if(j == 1) {
+                price += Math.min(helper(costs, i+1, 0), helper(costs, i+1, 2));
+            }
+            else {
+                price += Math.min(helper(costs, i+1, 0), helper(costs, i+1, 1));
+            }
+        }
+        
+        return price;
+    }
+
+    // Time Complexity : O(2^n)
```

Comment:
Please check time complexity here. We have two factors that affect it, coin denominations and amount